src/main/java/chess/domain/position/Position.java

```diff
@@ -0,0 +1,123 @@
+        for (File file : File.values()) {
+            putAllRankPositionToPool(file);
+        }
+    }
+
+    private static void putAllRankPositionToPool(File file) {
+        for (Rank rank : Rank.values()) {
+            POSITION_POOL.put(toKey(file, rank), new Position(file, rank));
+        }
+    }
+
+    private static String toKey(File file, Rank rank) {
+        return file.name() + rank.name();
+    }
+
+    private final File file;
+
+    private final Rank rank;
+
+    private Position(File file, Rank rank) {
+        this.file = file;
+        this.rank = rank;
+    }
+
+    public static Position of(File file, Rank rank) {
+        return POSITION_POOL.get(toKey(file, rank));
+    }
+
+    public Position createPositionByDifferencesOf(int fileDifference, int rankDifference) {
+        return Position.of(
+                file.createFileByDifferenceOf(fileDifference),
+                rank.createRankByDifferenceOf(rankDifference)
+        );
+    }
+
+    public boolean isOnSameRank(Position other) {
+        return rank == other.rank;
+    }
+
+    public boolean isOnSameFile(Position other) {
+        return file == other.file;
+    }
+
+    public boolean isOnPositiveSlopeDiagonal(Position other) {
+        int rankDifference = rank.subtract(other.rank);
+        int fileDifference = file.subtract(other.file);
+        return hasSameAbsoluteDifference(rankDifference, fileDifference) &&
+                hasSameSign(rankDifference, fileDifference);
+    }
+
+    public boolean isOnNegativeSlopeDiagonal(Position other) {
+        int rankDifference = rank.subtract(other.rank);
+        int fileDifference = file.subtract(other.file);
+        return hasSameAbsoluteDifference(rankDifference, fileDifference) &&
+                hasDifferentSign(rankDifference, fileDifference);
+    }
+
+    public boolean hasLowerFileThan(Position other) {
+        return file.subtract(other.file) < 0;
+    }
+
+    public boolean hasHigherFileThan(Position other) {
+        return file.subtract(other.file) > 0;
+    }
+
+    public boolean hasLowerRankThan(Position other) {
+        return rank.subtract(other.rank) < 0;
+    }
+
+    public boolean hasHigherRankThan(Position other) {
+        return rank.subtract(other.rank) > 0;
+    }
+
+    private boolean hasDifferentSign(int rankDifference, int fileDifference) {
+        return rankDifference * fileDifference < 0;
+    }
+
+    private boolean hasSameSign(int rankDifference, int fileDifference) {
+        return rankDifference * fileDifference >= 0;
+    }
+
+    private boolean hasSameAbsoluteDifference(int rankDifference, int fileDifference) {
+        return Math.abs(rankDifference) == Math.abs(fileDifference);
+    }
+
+    public boolean isOnKnightRoute(Position other) {
+        return Math.abs(rank.subtract(other.rank) * file.subtract(other.file)) == KNIGHT_CALCULATE_NUMBER;
+    }
+
+    public boolean isNotEquals(Position other) {
+        return !this.equals(other);
```

Comment:
```suggestion
        return !.equals(other);
```
클래스 내에서 `this`를 사용하는 호기의 기준이 있을까요?

Reply:
자바 문법에 서툴렀던 것 같습니다..!
인스턴스 자신과 `other`를 비교하기 때문에 `this.equals(other)` 이라고 기술 했는데 this를 안써도 되는지 몰랐습니다..!
수정하겠습니다!